src/Validation/Validation.php

```diff
@@ -308,6 +308,35 @@ public static function compareWith($check, $field, $context)
     }
 
     /**
+     * Checks if a string contains one or more non-alphanumeric characters.
+     *
+     * Returns true if string contains at least the specified number of non-alphanumeric characters
+     *
+     * $check can be passed as an array:
+     * ['check' => 'valueToCheck'];
+     *
+     * @param string|array $check Value to check
+     * @param int $count Number of non-alphanumerics to check for
+     * @return bool Success
+     */
+    public static function containsNonAlphaNumeric($check, $count = 1)
+    {
+        if (is_array($check)) {
+            extract(static::_defaults($check));
+        }
```

Comment:
To be honest I:

- just copied that from the [alphaNumerics() rule](https://github.com/cakephp/cakephp/blob/master/src/Validation/Validation.php#L106).  
- hesitated to even add array support since others like `url` only support strings which seemed enough for me

Some guidance would be welcome